yt_dlp/networking/common.py

```diff
@@ -74,6 +75,22 @@ def add_handler(self, handler: RequestHandler):
         assert isinstance(handler, RequestHandler), 'handler must be a RequestHandler'
         self.handlers[handler.RH_KEY] = handler
 
+    def get_handlers(self, filters=None):
+        """Return filtered handlers
+        @param filters: list of filters in the form of func(key, value) -> bool
+        """
+        if not filters:
+            filters = []
+        return dict(filter(lambda x: all(f(x[0], x[1]) for f in filters), self.handlers.items()))
+
+    def collect_from_handlers(self, collect_func, filters=None):
+        """
+        Collects data from handlers
+        @param collect_func: function to collect data from a handler, in the form of func(handler) -> Iterable
+        @param filters: list of filters for get_handlers()
+        """
+        return list(itertools.chain.from_iterable(collect_func(rh) for rh in self.get_handlers(filters).values()))
```

Comment:
These are unnecessary imo. For example,
```py
        self._request_director.collect_from_handlers(
            lambda rh: [(*target, rh.RH_NAME) for target in rh.get_supported_targets()],
            [lambda _, v: isinstance(v, ImpersonateRequestHandler)]
        )
```
from YoutubeDL can just be written as
```py
        [[(*target, rh.RH_NAME) for target in rh.get_supported_targets()]
         for rh in self._request_director.handlers.values()
         if isinstance(rh, ImpersonateRequestHandler)]
```
which is shorter, easier to read, and more pythonic

Reply:
fair enough. can always generalise later if there is more use cases